health-sdk/sdk/src/main/java/net/gini/android/health/sdk/review/ReviewViewModel.kt

```diff
@@ -295,12 +294,26 @@ internal class ReviewViewModel(val giniHealth: GiniHealth, val configuration: Re
     internal fun getFileAsByteArray(externalCacheDir: File?) {
         viewModelScope.launch {
             withContext(Dispatchers.IO) {
-                val paymentRequest = async { getPaymentRequest() }.await()
+                sendFeedbackAndStartLoading()
+                val paymentRequest = try {
+                    getPaymentRequest()
+                } catch (throwable: Throwable) {
+                    GiniHealth.PaymentState.Error(throwable)
+                    _paymentNextStep.tryEmit(PaymentNextStep.OpenSharePdfError(throwable.message ?: "Error getting payment request"))
+                    return@withContext
+                }
                 val byteArrayResource = async {  giniHealth.giniHealthAPI.documentManager.getPaymentRequestDocument(paymentRequest.id) }.await()
                 when (byteArrayResource) {
-                    is Resource.Cancelled -> throw Exception("Cancelled")
-                    is Resource.Error -> throw Exception(byteArrayResource.exception)
+                    is Resource.Cancelled -> {
+                        GiniHealth.PaymentState.Error(Exception("Cancelled"))
```

Comment:
[Request] This event needs to be emitted on `giniHealth.setOpenBankState()` so that clients can receive it. It is also collected by the ReviewFragment and I have a proposal how to differentiate in the ReviewFragment between "payment request error" and "pdf error" and avoid showing both:
- Emit all pdf errors as a `GiniHealth.PaymentState.Error`, but use a custom Exception subclass, for example `OpenSharedPdfException`. This way we have one error channel.
- Check the instance of the Throwable in ReviewFragment's `handlePaymentState()` method and show the pdf error message if it's the `OpenSharedPdfException`.

Reply:
As discussed, for now both `payment request` and `get pdf` will display the same error and follow the same steps after tapping on `Retry`. Should we need separate error messages, an `OpenSharedPdfException` will be introduced to allow us to differentiate between the two.